data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/consumer/KafkaSourceCustomConsumer.java

```diff
@@ -121,29 +126,21 @@ private AcknowledgementSet createAcknowledgementSet(Map<TopicPartition, Range<Lo
                         try {
                             int partitionId = partition.partition();
                             if (!partitionCommitTrackerMap.containsKey(partitionId)) {
-                                OffsetAndMetadata committedOffsetAndMetadata = null;
-                                synchronized(consumer) {
-                                    committedOffsetAndMetadata = consumer.committed(partition);
-                                }
+                                OffsetAndMetadata committedOffsetAndMetadata = consumer.committed(partition);
                                 Long committedOffset = Objects.nonNull(committedOffsetAndMetadata) ? committedOffsetAndMetadata.offset() : null;
                                 partitionCommitTrackerMap.put(partitionId, new TopicPartitionCommitTracker(partition, committedOffset));
                             }
                             OffsetAndMetadata offsetAndMetadata = partitionCommitTrackerMap.get(partitionId).addCompletedOffsets(offsetRange);
                             updateOffsetsToCommit(partition, offsetAndMetadata);
                         } catch (Exception e) {
-                            LOG.error("Failed to seek to last committed offset upon positive acknowledgement "+partition, e);
+                            LOG.error("Failed to seek to last committed offset upon positive acknowledgement {}", partition, e);
                         }
                     });
                 } else {
                     negativeAcknowledgementSetCounter.increment();
                     offsets.forEach((partition, offsetRange) -> {
-                        try {
-                            synchronized(consumer) {
-                                OffsetAndMetadata committedOffsetAndMetadata = consumer.committed(partition);
-                                consumer.seek(partition, committedOffsetAndMetadata);
-                            }
-                        } catch (Exception e) {
-                            LOG.error("Failed to seek to last committed offset upon negative acknowledgement "+partition, e);
+                        synchronized(partitionsToReset) {
+                            partitionsToReset.add(partition);
```

Comment:
This seems a nice change. Did you do any performance testing or find any particular pitfalls with the original approach?

Reply:
I haven't done any performance testing. But definitely having a lock every time consumer is accessed is not good.